e2e/runner/setup_ton.go

```diff
@@ -103,6 +101,11 @@ func (r *E2ERunner) ensureTONChainParams(gw *ton.AccountInit) error {
 		return errors.Wrap(err, "unable to broadcast TON chain params tx")
 	}
 
+	resetMsg := observertypes.NewMsgResetChainNonces(creator, chainID, 0, 0)
```

Comment:
why do we reset nonces explicitly only for TON?

Reply:
We have the same logic for SOL btw. Reason:
```
3:26PM ERR tx post processing failed 
error="EVM hook keeper.Hooks failed: failed to process ZEVM inbound V1: identifiers: 
0x65a45c57636f9BcCeD4fe193A602008578BcA90b-101-2015142-0 (chain \"ton_localnet\"):
cannot find receiver chain nonce" module=x/evm
```

Reply:
Any chains that are not configured at genesis needs their nonces reset. See https://github.com/zeta-chain/node/pull/2990